examples/reference/widgets/FileInput.ipynb

```diff
@@ -146,101 +265,221 @@
-    "2. BokehJS uses the [FileReader.readAsDataURL](https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsDataURL) method to encode the file as a data URL; browsers can [limit](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs#common_problems) the length of that URL\n",
-    "3. BokehJS (or the libraries that it uses) may fail at creating the message from a very (very!) large string\n",
-    "4. `Tornado` imposes two limits on the data being transferred: (1) on the maximum size of a buffer and (2) on the maximum size of the websocket message. (1) is 100 MB by default (set by Tornado) and (2) is 20 MB by default (set by Bokeh).\n",
+    "Understanding how file uploads work helps you make informed decisions about size limits and user experience:\n",
     "\n",
-    "#### Increase the limits\n",
+    "**The journey of an uploaded file:**\n",
     "\n",
-    "While there's nothing much that can be done about 1., 2. and 3. (except informing your users), the limits defined by Tornado and Bokeh can be overwritten.\n",
+    "1. **Browser loading** - File content is loaded into browser memory\n",
+    "2. **Encoding** - Content is converted to [base64](https://en.wikipedia.org/wiki/Base64) format (making it ~33% larger)\n",
+    "3. **Message creation** - The encoded data is packaged into a JSON message\n",
+    "4. **WebSocket transmission** - Data travels through [WebSocket](https://en.wikipedia.org/wiki/WebSocket) connection via [Tornado](https://www.tornadoweb.org/en/stable/), the network and other servers.\n",
+    "5. **Backend processing** - Your Python application receives the data as a `bytes` `value`.\n",
     "\n",
-    "##### Server context\n",
+    "#### Technical Constraints\n",
     "\n",
-    "In a server context your application must be executed with `python your_app.py` (because `panel serve` doesn't allow to configure all the options provided by Bokeh and Tornado):\n",
+    "Several layers impose size limitations:\n",
+    "\n",
+    "1. **Browser limits** - Generally several GB, rarely the bottleneck\n",
+    "2. **WebSocket message size** - Default: 20 MB (Tornado Bokeh setting)\n",
+    "3. **Buffer size** - Default: 100 MB (Tornado setting)\n",
+    "4. **Encoding overhead** - Base64 increases size by ~33%\n",
+    "5. **Network/ infrastructure/ Servers** - For example if you are running your app or JupyterHub on Kubernetes, Kubernetes sets its own limits.\n",
+    "\n",
+    "#### Configuring Higher Limits\n",
+    "\n",
+    "**For server deployments,** you can increase limits when launching your application:\n",
     "\n",
     "```python\n",
-    "# your_app.py\n",
+    "# production_app.py\n",
     "import panel as pn\n",
     "\n",
+    "# Your application code here\n",
     "app = ...\n",
     "\n",
+    "# Configure for larger uploads (150 MB example)\n",
     "MAX_SIZE_MB = 150\n",
     "\n",
     "pn.serve(\n",
```

Comment:
Document how to set this when using `panel serve`. I see `--websocket-max-message-size` but no `pn.config` parameter. I don't see any way to set `http_server_kwargs`.

Reply:
Why can't Panel/ Bokeh websocket comms just automatically chunk and do what is needed to avoid problems?